packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/DKGService.java

```diff
@@ -0,0 +1,78 @@
+package software.uncharted.terarium.hmiserver.service.data;
+
+import co.elastic.clients.elasticsearch._types.query_dsl.Query;
+import co.elastic.clients.elasticsearch._types.query_dsl.QueryBuilders;
+import co.elastic.clients.elasticsearch.core.SearchRequest;
+import co.elastic.clients.elasticsearch.core.search.SourceConfig;
+import io.micrometer.observation.annotation.Observed;
+import java.io.IOException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import software.uncharted.terarium.hmiserver.configuration.ElasticsearchConfiguration;
+import software.uncharted.terarium.hmiserver.models.mira.DKG;
+import software.uncharted.terarium.hmiserver.service.elasticsearch.ElasticsearchService;
+
+@Service
+@RequiredArgsConstructor
+public class DKGService {
+
+	private final ElasticsearchConfiguration elasticConfig;
+	private final ElasticsearchService elasticService;
+	private static final String EMPHASIS = "^2";
+
+	@Observed(name = "function_profile")
+	public List<DKG> searchEpiDKG(final Integer page, final Integer pageSize, final String q, final SourceConfig source)
+		throws IOException {
+		return searchDKG(elasticConfig.getEpiDKGIndex(), page, pageSize, q, source);
+	}
+
+	@Observed(name = "function_profile")
+	public List<DKG> searchClimateDKG(
+		final Integer page,
+		final Integer pageSize,
+		final String q,
+		final SourceConfig source
+	) throws IOException {
+		return searchDKG(elasticConfig.getClimateDKGIndex(), page, pageSize, q, source);
+	}
+
+	private List<DKG> searchDKG(
+		final String index,
+		final Integer page,
+		final Integer pageSize,
+		final String q,
+		final SourceConfig source
+	) throws IOException {
+		final SearchRequest.Builder builder = new SearchRequest.Builder().index(index).from(page).size(pageSize);
+
+		if (q != null && !q.isEmpty()) {
+			Query query = QueryBuilders.multiMatch().fields(DKG.NAME + EMPHASIS, DKG.DESCRIPTION).query(q).build()._toQuery();
```

Comment:
Maybe better if "emphasis" is a function/method to make things a little more future proof?

```
private static String emphasis(String s, int boost) {
  return s + "^" + String.valueOf(boost);
}
```

Reply:
oh true!